HW4/HW4.py

```diff
@@ -0,0 +1,191 @@
+import math
+
+import numpy as np
+import pandas
+import plotly.figure_factory as ff
+
+# import pydot
+import statsmodels.api
+
+# from matplotlib import pyplot as plt
+# from pandas import DataFrame
+from plotly import express as px
+from plotly import graph_objects as go
+
+# import matplotlib.pyplot as plt
+from plotly.subplots import make_subplots
+from scipy import stats
+
+# from io import StringIO
+
+# from sklearn import datasets, tree
+# from sklearn.metrics import confusion_matrix
+# from sklearn.model_selection import GridSearchCV
+# from sklearn.tree import DecisionTreeClassifier, export_graphviz
+
+# import tempfile
+# from PIL import Image
+
```

Comment:
You need to use the `def main()` function and `if __name__ == "__main__":`

Reply:
I updated it. Thanks!